indra/llprimitive/tests/llgltfmaterial_test.cpp

```diff
@@ -0,0 +1,255 @@
+            {
+                LLGLTFMaterial material;
+                material.setDoubleSided(doubleSided, forOverride);
+                const bool overrideBit = (doubleSided == false) && forOverride;
+                ensure_equals("LLGLTFMaterial: double sided = " + std::to_string(doubleSided) + " override bit when forOverride = " + std::to_string(forOverride), material.mOverrideDoubleSided, overrideBit);
+                ensure_gltf_material_serialize("double sided = " + std::to_string(doubleSided), material);
+            }
+        }
+
+        for (LLGLTFMaterial::AlphaMode alphaMode : alphaModes)
+        {
+            for (bool forOverride : forOverrides)
+            {
+                LLGLTFMaterial material;
+                material.setAlphaMode(alphaMode, forOverride);
+                const bool overrideBit = (alphaMode == LLGLTFMaterial::ALPHA_MODE_OPAQUE) && forOverride;
+                ensure_equals("LLGLTFMaterial: alpha mode = " + std::to_string(alphaMode) + " override bit when forOverride = " + std::to_string(forOverride), material.mOverrideAlphaMode, overrideBit);
+                ensure_gltf_material_serialize("alpha mode = " + std::to_string(alphaMode), material);
+            }
+        }
+    }
+
+    // Test that a GLTF material's transform components serialize as expected
+    template<> template<>
+    void llgltfmaterial_object_t::test<4>()
+    {
+        LLGLTFMaterial material;
+        LLGLTFMaterial::TextureTransform& transform = material.mTextureTransform[LLGLTFMaterial::GLTF_TEXTURE_INFO_BASE_COLOR];
+        transform.mOffset[VX] = 1.f;
+        transform.mOffset[VY] = 2.f;
+        transform.mScale[VX] = 0.05f;
+        transform.mScale[VY] = 100.f;
+        transform.mRotation = 1.571f;
+        ensure_gltf_material_serialize("material with transform", material);
+    }
+
+    // Test that a GLTF material avoids serializing a material unnecessarily
+    template<> template<>
+    void llgltfmaterial_object_t::test<5>()
+    {
+        {
+            const LLGLTFMaterial material;
+            const std::string material_json = material.asJSON();
+            ensure_gltf_material_trimmed(material_json, "pbrMetallicRoughness");
+            ensure_gltf_material_trimmed(material_json, "normalTexture");
+            ensure_gltf_material_trimmed(material_json, "emissiveTexture");
+            ensure_gltf_material_trimmed(material_json, "occlusionTexture");
+        }
+
+        {
+            LLGLTFMaterial metallic_factor_material;
+            metallic_factor_material.setMetallicFactor(0.5);
+            const std::string metallic_factor_material_json = metallic_factor_material.asJSON();
+            ensure_gltf_material_trimmed(metallic_factor_material_json, "baseColorTexture");
+            ensure_gltf_material_trimmed(metallic_factor_material_json, "metallicRoughnessTexture");
+        }
+    }
+
+    // Test that a GLTF material preserves values on serialization
```

Comment:
If you are testing serialiation, I suggest enshuring that 'default' values do not get serialized after variable was set to something then reset to default.